sql/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -118,6 +182,8 @@ public FunctionImplementation getForTypes(List<DataType> argumentTypes) throws I
             FunctionInfo info = new FunctionInfo(new FunctionIdent(NAME, argumentTypes), DataTypes.BOOLEAN);
             if (leftType.id() == ObjectType.ID && rightType.id() == ObjectType.ID) {
                 return new ObjectEqOperator(info);
+            } else if (leftType.id() == IpType.ID || rightType.id() == IpType.ID) {
```

Comment:
I think you'll also have to override and adapt the `getSignature` method, so that the cast to `[ip, ip]` can be by-passed.

Reply:
Aha!, good hint, will explore this avenue.